### DISCUSSION TITLE
Negative distance to my device

### DISCUSSION THREAD

**Discussion by felix-1234** [mKeRix/room-assistant#660] [Q&A]
Hi everybody!

New here and new user of RA.

Just installed RA on a PiZeroW using BT Classic and integrates with HA and everything works nicely. My two entities shows up in HA but the only thing is if I'm approx 1m from my Pi with my watch that I'm tracking it says that I'm ca -2m away.

What is the best way to tweak this so I can get a more accurate reading?

`bluetoothClassic:
  minRssi: -10
  hciDeviceId: 0
  addresses:
    - '00:00:00:00:00:00' # F Apple Watch
    - '00:00:00:00:00:00' # L Apple Watch
`

Any tips? :)

Thanks for a great software, this will help my automations a lot.

Regards

// Fredrik

**Comment by haltug** [comment]
Same for me, would be interesting to know how to solve this.

**Comment by mKeRix** [comment]
For Bluetooth Classic the distance is actually not in meters at all, it's just some arbitrary value that scales with signal strength (and therefore also device distance). If you really need a meter estimation of distance I'd suggest to either check if you can use BLE instead or build a template sensor that shifts and factors the value into something that estimates proper meters in Home Assistant. BLE works for many beacons, tags, Androids via the Home Assistant companion and iPhones via the room-assistant companion app.
